spec-tests/src/test/java/org/candlepin/spec/pools/PostBindBonusPoolSpecTest.java

```diff
@@ -0,0 +1,1198 @@
+
+        ProductDTO unlimitedVirtProd = createUnlimitedVirtProduct(adminClient, ownerKey);
+        PoolDTO unlimitedMasterPool = Pools.randomUpstream(unlimitedVirtProd).quantity(10L);
+        unlimitedMasterPool = adminClient.owners().createPool(ownerKey, unlimitedMasterPool);
+        PoolDTO unlimitedBonusPool =
+            getBonusPool(adminClient, ownerKey, unlimitedMasterPool.getSubscriptionId());
+
+        guestClient.consumers().bindPool(guest.getUuid(), unlimitedMasterPool.getId(), 1);
+        unlimitedBonusPool = adminClient.pools().getPool(unlimitedBonusPool.getId(), null, null);
+        assertEquals(-1, unlimitedBonusPool.getQuantity());
+        unlimitedMasterPool = adminClient.pools().getPool(unlimitedMasterPool.getId(), null, null);
+        assertEquals(1, unlimitedMasterPool.getConsumed());
+        assertEquals(0, unlimitedMasterPool.getExported());
+
+        // even if one quantity was consumed but not exported, do not update quantity of the bonus pool
+        systemClient.consumers().bindPool(systemUser.getUuid(), unlimitedMasterPool.getId(), 9);
+        unlimitedMasterPool = adminClient.pools().getPool(unlimitedMasterPool.getId(), null, null);
+        assertEquals(unlimitedMasterPool.getConsumed(), unlimitedMasterPool.getQuantity());
+        assertNotEquals(unlimitedMasterPool.getExported(), unlimitedMasterPool.getConsumed());
+        unlimitedBonusPool = adminClient.pools().getPool(unlimitedBonusPool.getId(), null, null);
+        assertEquals(-1, unlimitedBonusPool.getQuantity());
+    }
+
+    @Test
+    void shouldNotChangeHostlimitedEntitlementDerivedPoolQuantityWhenSourceEntitlementsQuantityIsReduced()
+        throws Exception {
+        ApiClient adminClient = ApiClients.admin();
+        OwnerDTO owner = adminClient.owners().createOwner(Owners.random());
+        String ownerKey = owner.getKey();
+
+        ConsumerDTO systemUser = Consumers.random(owner, ConsumerTypes.System);
+        systemUser.setUsername("admin");
+        systemUser = adminClient.consumers().createConsumer(systemUser, "admin",
+        owner.getKey(), null, true);
+        ApiClient systemClient = ApiClients.trustedConsumer(systemUser.getUuid());
+
+        ProductDTO hostLimitedProd = createHostLimitedProduct(adminClient, ownerKey);
+        PoolDTO hostLimitedMasterPool = Pools.randomUpstream(hostLimitedProd).quantity(10L);
+        hostLimitedMasterPool = adminClient.owners().createPool(ownerKey, hostLimitedMasterPool);
+
+        int initialPoolSize = adminClient.owners().listOwnerPools(ownerKey).size();
+        JsonNode ent = systemClient.consumers()
+            .bindPool(systemUser.getUuid(), hostLimitedMasterPool.getId(), 4).get(0);
+        List<PoolDTO> pools = adminClient.owners().listOwnerPools(ownerKey);
+        assertEquals(initialPoolSize + 1, pools.size());
+
+        List<PoolDTO> entDerivedPools = pools.stream()
+            .filter(pool -> pool.getType().equals("ENTITLEMENT_DERIVED"))
```

Comment:
Just a note. Generally it is better to reverse such conditions to avoid NPE.

```java
"ENTITLEMENT_DERIVED".equals(pool.getType())
```

Reply:
Good call. Updated